### PR TITLE
Hardcode/template the hostname in dhclient.conf

### LINKED ISSUE #136: dhclient should not use gethostname() but a hardcoded (templated) value
It happens that the local router/DHCPd doesn't provide a `domain-name` value.
```
$ hostname -f
hostname: Name or service not known
```
The `/etc/dhcp/dhclient.conf` configuration uses `gethostname()`, which wants to return a FQDN, which fails (because there isn't any).

This is why we don't always have a `client-hostname` in tincmaster's DHCP leases.

We don't have control over the local DHCPd, so our only solution is to hardcode `send host-name` in `/etc/dhcp/dhclient.conf`, templating it using `{{ idc_uniq_device_name |replace("_", "-") }}` . This has to be handled by the `rename_device` role and friends.

### PR DESCRIPTION
**Pull request by letompouce**
Fixes #136

### CODE REVIEW

**Review by swbsf — COMMENTED** [review]
Wouldn't it be cleaner to make sure gethostname() actually returns something?

**Comment by letompouce** [comment]
I hardly see how faking a domain would be considered cleaner but I'm all open to suggestions :)

**Review by swbsf — APPROVED** [review]

File: roles/system/templates/dhclient.conf.j2 [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+#	man page for more information about the syntax of this file
+#	and a more comprehensive list of the parameters understood by
+#	dhclient.
+#
+# Normally, if the DHCP server provides reasonable information and does
+#	not leave anything out (like the domain name, for example), then
+#	few changes must be made to this file, if any.
+#
+
+option rfc3442-classless-static-routes code 121 = array of unsigned integer 8;
```
Comment (fheslouin) [inline roles/system/templates/dhclient.conf.j2]: At this stage you have to use {{ ansible_hostname }}, {{ idc_uniq_device_name }} is only used on "master" tag

File: roles/system/templates/dhclient.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+send host-name "{{ idc_uniq_device_name |replace("_", "-") }}";
```
Comment (fheslouin) [inline roles/system/templates/dhclient.conf.j2]: At this stage you have to use {{ ansible_hostname }}, {{ idc_uniq_device_name }} is only used on "master" tag

**Review by fheslouin — APPROVED** [review]

**Pull request merged**